pubsub/cloud-client/quickstart/pub_test.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pytest
+
+from google.cloud import pubsub_v1
+
+import pub
+
+PROJECT = os.environ['GCLOUD_PROJECT']
+TOPIC = 'quickstart-pub-test-topic'
+
+
+@pytest.fixture(scope='module')
+def publisher_client():
+    yield pubsub_v1.PublisherClient()
+
+
+@pytest.fixture(scope='module')
+def topic(publisher_client):
+    topic_path = publisher_client.topic_path(PROJECT, TOPIC)
+
+    try:
```

Comment:
Why not continue if the topic already exists and delete after the yield? This test will leave a resource on the project after the test runs otherwise.

Reply:
I added a cleanup section in test. Is it what you mean by "delete after the yield"?

Reply:
No. If the test fails, then the cleanup won't run.

See

https://github.com/googleapis/google-cloud-python/blob/0ba121b5646ce2bc1537ac8be99ad87461a15a06/bigquery/docs/snippets.py#L86-L96

as an example. The fixture yields an object, then the cleanup code runs after the tests finish.

Reply:
That makes sense. This is cool. 
Done.